src/plays/password-generator/PasswordGenerator.jsx

```diff
@@ -0,0 +1,199 @@
+import { getPlayById } from "meta/play-meta-util";
+
+import PlayHeader from "common/playlists/PlayHeader";
+import { useEffect, useState } from "react";
+
+import "./password-generator-style.css";
+import data from "./data.json";
+
+const config = {
+  length: 12,
+  numbers: true,
+  special: true,
+  uppercase: true,
+  lowercase: true,
+  // excludeSimilarCharacters: true,
+};
+
+function PasswordGenerator(props) {
+  // Do not remove the below lines.
+  // The following code is to fetch the current play from the URL
+  const { id } = props;
+  const play = getPlayById(id);
+
+  // Your Code Start below.
+  const [password, setPassword] = useState({ status: false, password: "" });
+  const [passwordConfig, setPasswordConfig] = useState({ ...config });
+  const [error, setError] = useState(false);
+
+  // generating lowercase characters
+  data.lowercase = data?.uppercase?.map((i) => i.toLowerCase());
+
+  // generate a random number within limit which is provided
+  const randomNumberGenerator = (limit) => {
+    let result = 0;
+    while (limit) {
+      result = Math.floor(Math.random() * Math.floor(Math.random() * 100));
+      if (result < limit) return result;
+      continue;
+    }
+  };
+
+  // arrange data to feed the generatePassword function
+  const arrangeData = () => {
+    const { numbers, special, uppercase, lowercase } = passwordConfig;
+
+    const parseData = (val, key) => {
+      if (val) return data[key];
+      return [];
+    };
+
+    const concated = [
+      ...parseData(numbers, "numbers"),
+      ...parseData(special, "special"),
+      ...parseData(uppercase, "uppercase"),
+      ...parseData(lowercase, "lowercase"),
+    ];
+    return concated;
+  };
+
+  // Generate a random password
+  const generatePassword = () => {
+    setError(false)
+    const concated = arrangeData();
+    let finalPassword = "";
+    for (let i = 0; i < passwordConfig.length; i++) {
+      finalPassword += concated[randomNumberGenerator(concated.length)];
+    }
+    return finalPassword;
+  };
+
+  // generate password button click handler
+  const generateHander = () => {
+    const finalPassword = generatePassword();
+    setPassword({ status: false, password: finalPassword });
+  };
+
+  // handling checkbox and updating the config
+  const handleCheckedItem = (name, checked) => {
+    const modifiedConfig = { ...passwordConfig };
+    delete modifiedConfig.length
```

Comment:
In some lines ; are missing. It good to be uniform.

Reply:
yes. semicolons are added